contracts/std-lib-extended/src/bytes.sw

```diff
@@ -44,6 +44,24 @@ impl EvmAddress {
     }
 }
 
+/// The number of bytes in a B512.
+pub const B512_BYTE_COUNT: u64 = 64u64;
+
+impl B512 {
+    /// Returns a pointer to the B512's packed bytes.
+    fn packed_bytes(self) -> raw_ptr {
+        __addr_of(self.bytes)
+    }
+
+    /// Gets a B512 from a pointer to packed bytes.
+    fn from_packed_bytes(ptr: raw_ptr) -> Self {
+        let component_0 = asm(ptr: ptr) { ptr: b256 };
+        let component_1 = asm(ptr: ptr.add_uint_offset(B256_BYTE_COUNT)) { ptr: b256 };
```

Comment:
do we really need asm here :P

Reply:
oops great catch, meant to change this but forgot to